### PR TITLE
Feature/setup

### PR DESCRIPTION
**Pull request by Salimer**
## What does this PR do?

This PR is raised to implement the following features 👍:

- [x] Build a To-Do app using React and Vite
- [x] The app adds, removes,s and edits new tasks
- [x] Use react icons 
- [x] Save tasks in local storage
- [x] Use different hooks to improve performance

## Description of the completed tasks 🙌

- [x] Added the HTML for the structure. 
- [x] Added the CSS file for the styling.
- [x] Added the js components for React.
- [x] Used the linter tool that analyzes my source code to flag programming errors, bugs, stylistic errors, and suspicious constructs. 💯

### CODE REVIEW

File: murple_logo.png
Comment (Mar12358) [inline murple_logo.png:1]: Nice Logo! 🥇

File: README.md
Comment (Mar12358) [inline README.md:1]: Very Good Job on your Documentation, Congratulations! 💯

File: src/components/InputTodo.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+</button>
+
+    </form>
+    <span className="submit-warning">{message}</span>
+        </>
+    )
+  };
+  InputTodo.propTypes = {
+    addTodoItem: PropTypes.func.isRequired
+  }
```
Comment (Mar12358) [inline src/components/InputTodo.jsx:39]: Good Job!

File: src/components/InputTodo.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { useState } from "react";
+import { FaPlusCircle } from "react-icons/fa";
```
Comment (Mar12358) [inline src/components/InputTodo.jsx:2]: Good Job implementing react-icons! 🥇

**Review by Mar12358 — CHANGES_REQUESTED** [review]
## Hi @Salimer !
You did a great job so far, but there are still some required sections missing.

### Highlights 💯 
✔️ The repo is professional according to the [basic rules]
✔️ Descriptive README file.
✔️ Meaningful Commit history.
✔️ Professional PR
✔️ Check linters
✔️ Descriptive Branch name.
✔️ App works as expected.
✔️ App runs successfully.
✔️ Hooks (useEffect, useState)
✔️ Props and state usage.
✔️ Good job using Vite.
✔️ Good quality code (DRY, KISS, and YAGNI).
almost-there
## Required Changes ♻️ ♻️ ♻️
You did a great effort on your app functionality and its style, but there are still some required concepts on the tutorial that are not implemented on your project. Please check the following checklist and the comments below:

- [ ] ❌ Gitflow [OPTIONAL] Remember to have a dev branch parallel to main, and the feature branch that you actually code. Then compare from your feature branch to develop branch with your PR. Remember using it further on.
 
- [ ] I kindly suggest adding routing with React Router, since it's one of the required sections of the tutorial. (Part 9)

- [ ] Please, consider adding the navbar, since it is a required section in the tutorioal. This would lead to implement its features as well, such as Login/logout components, Home/About/Login/Profile pages, routing in general, limiting edit button access and profile preview if not logged in.


## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me @Mar12358 when you comment so I can receive the notification or contact me on Slack.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Salimer** [comment]
Thank you @Mar12358 for the thorough review. By mistake I started this pull request to the wrong branch 'main' when I should make it to the 'development' branch. I will address the suggested changes in another pull request. Please review my new [Pull Request](https://github.com/Salimer/react-to-do-list/pull/2)

**Review by Mar12358 — APPROVED** [review]
Hi @Salimer 👋,

## Status: Approved 🟢   
well-1
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

### Highlights
✔️ The repo is professional according to the [basic rules]
✔️ Descriptive README file.
✔️ Meaningful Commit history.
✔️ Professional PR
✔️ Check linters
✔️ Descriptive Branch name.
✔️ App works as expected.
✔️ App runs successfully.
✔️ Hooks (useEffect, useState)
✔️ Props and state usage.
✔️ Good job using Vite.
✔️ Good quality code (DRY, KISS, and YAGNI).
✔️ **Good Job implemented the required changes**
✔️ Routing
✔️ Add navbar
✔️ GitFlow

### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Mar12358 

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Salimer** [comment]
Thank you @Mar12358 for the thorough review 🥳

**Pull request closed without merging**